concordium-consensus/src/Concordium/GlobalState/Persistent/Account/StructureV1.hs

```diff
@@ -0,0 +1,1314 @@
+                  _delegationTarget = paseDelegatorTarget,
+                  _delegationPendingChange = PendingChangeEffectiveV1 <$> paseDelegatorPendingChange,
+                  ..
+                }
+
+-- |Migrate a 'PersistentAccountStakeEnduring' from one blob store to another.
+migratePersistentAccountStakeEnduring ::
+    (SupportMigration m t, IsAccountVersion av) =>
+    PersistentAccountStakeEnduring av ->
+    t m (PersistentAccountStakeEnduring av)
+migratePersistentAccountStakeEnduring PersistentAccountStakeEnduringNone =
+    return PersistentAccountStakeEnduringNone
+migratePersistentAccountStakeEnduring PersistentAccountStakeEnduringBaker{..} = do
+    newBakerInfo <- migrateReference return paseBakerInfo
+    return $!
+        PersistentAccountStakeEnduringBaker
+            { paseBakerInfo = newBakerInfo,
+              ..
+            }
+migratePersistentAccountStakeEnduring PersistentAccountStakeEnduringDelegator{..} =
+    return $! PersistentAccountStakeEnduringDelegator{..}
+
+
+-- |This relies on the fact that the 'AccountV2' hashing of 'AccountStake' is independent of the
+-- staked amount.
+instance (MonadBlobStore m) => MHashableTo m (AccountStakeHash 'AccountV2) (PersistentAccountStakeEnduring 'AccountV2) where
+    getHashM stake = getHash <$> persistentToAccountStake stake 0
+
+-- * Enduring account data
+
+-- |Enduring data associated with an account. This is data that does not change very often.
+-- The 'AccountMerkleHash' is computed and stored for this data so that we can avoid deserializing
+-- elements of it unnecessarily.
+--
+-- The persisting account data is stored under an 'EagerBufferedRef' as this includes the account
+-- keys, which are required for any transaction originating from the account, and so are loaded
+-- eagerly.
+--
+-- The encrypted amount is 'Nullable' so that accounts with no encrypted balance can be represented
+-- succinctly.  The encrypted balance is stored as a 'LazyBufferedRef', which is not automatically
+-- cached with the account, since loading the encrypted balance is relatively expensive and likely
+-- not necessary for many operations.
+--
+-- The release schedule is likewise 'Nullable' and stored under a 'LazyBufferedRef' for similar reasons.
+-- However, as well as the 'LazyBufferedRef', we store the locked balance (if it is non-trivial).
+-- This is because we typically require ready access to the locked balance so that we can compute
+-- the available balance.
+--
+-- The stake is not stored under a reference (excepting the baker info, as per the definition of
+-- 'PersistentAccountStakeEnduring'). This is since the information is relatively succinct.
+data PersistentAccountEnduringData (av :: AccountVersion) = PersistentAccountEnduringData
+    { -- |The Merkle hash computed from the other fields.
+      paedHash :: !(AccountMerkleHash av),
+      -- |A reference to the persisting account data.
+      paedPersistingData :: !(EagerBufferedRef PersistingAccountData),
```

Comment:
PersistingAccountData has two copies of account keys. Would it not be better to also have a new version of that with a more refined representation of the accountinformation and credentials to remove duplication? RawAccountCredential does not have much more data, and we could probably just have that and not the account keys.

Having just account keys would be good as well, but then we'd have to have the ability to load them without loading everything else.

Reply:
I do not like this very much and I thought that was in scope.

But it is too late now, so OK.